format/tap/tap.go

```diff
@@ -54,9 +55,11 @@ func decodeTapBlock(d *decode.D) {
 	// read header, fragment, or data block
 	switch length {
 	case 0:
-		// fragment with no data
+		d.Fatalf("TAP fragments with 0 bytes are not supported")
 	case 1:
-		d.FieldRawLen("data", 8)
+		d.FieldStruct("data", func(d *decode.D) {
+			d.FieldRawLen("bytes", 8)
+		})
```

Comment:
Should be `d.FieldRawLen` again?

Reply:
Sorry, mis-read. Is the struct around it needed?

Reply:
I wanted to wrap each type of block to be able to distinguish between `header` and `data` blocks.

As a 1-byte data block there is no `flag` or `checksum` byte so this can not be passed to the `decodeDataBlock()` without that handling the length. So the struct is still useful to know the block type, and aid in JSON parsing.

I suppose another block type could be introduced, but that seems overkill for what is actually a rare occurrence.

Reply:
Aha i see, makes sense